core/node/node_sync/src/client.rs

```diff
@@ -97,8 +97,8 @@ impl MainNodeClient for Box<DynClient<L2>> {
     async fn fetch_protocol_version(
         &self,
         protocol_version: ProtocolVersionId,
-    ) -> EnrichedClientResult<Option<api::ProtocolVersion>> {
-        self.get_protocol_version(Some(protocol_version as u16))
+    ) -> EnrichedClientResult<Option<api::ProtocolVersionInfo>> {
+        self.get_protocol_version_info(Some(protocol_version as u16))
```

Comment:
Is it impossible to have an updated EN for a certain env with the main node not upgraded yet? If it *is* possible, `get_protocol_version_info()` will error. AFAIU, it's possible to use `get_protocol_version` as a fallback (`api:: ProtocolVersionInfo` is a subset of `api::ProtocolVersion`).

Reply:
> Is it impossible to have an updated EN for a certain env with the main node not upgraded yet?

I was operating under that assumption, yes. @perekopskiy do you think that's a fair assumption?

It would probably be a bit awkward to try and use jsonrpsee in the way you propose but I can use `get_protocol_version` as a fallback and add `impl From<api::ProtocolVersion> for api:: ProtocolVersionInfo`

Reply:
This assumption could be reasonable for outside EN operators, but IMO it's too shaky in the general case. E.g., in our infra, we have one of our ENs on the Era testnet and mainnet auto-updated from `main`.

Reply:
@slowli I have added a fallback just in case but I still suspect it is very unlikely to ever trigger since that can only happen during protocol upgrade. In any case PTAL